sklearn/linear_model/huber.py

```diff
@@ -0,0 +1,267 @@
+# Authors: Manoj Kumar mks542@nyu.edu
+# License: BSD 3 clause
+
+import numpy as np
+
+from scipy import optimize, sparse
+
+from ..base import BaseEstimator, RegressorMixin
+from .base import LinearModel
+from ..utils import check_X_y
+from ..utils import check_consistent_length
+from ..utils import axis0_safe_slice
+from ..utils.extmath import safe_sparse_dot
+
+
+def _huber_loss_and_gradient(w, X, y, epsilon, alpha, sample_weight=None):
+    """Returns the Huber loss and the gradient.
+
+    Parameters
+    ----------
+    w: ndarray, shape (n_features + 1,) or (n_features + 2,)
+        Feature vector.
+        w[:n_features] gives the feature vector
```

Comment:
I would say "is the" not "gives the". Also: feature vector? you mean coefficient, right?